packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3799,27 +3799,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
        && notification is! ScrollEndNotification) {
       return;
     }
-    if (notification is ScrollStartNotification
-       && _dataWhenToolbarShowScheduled != null) {
-      return;
-    }
-    if (notification is ScrollEndNotification
-       && _dataWhenToolbarShowScheduled == null) {
-      return;
-    }
-    if (notification is ScrollEndNotification
-       && _dataWhenToolbarShowScheduled!.value != _value) {
-      _dataWhenToolbarShowScheduled = null;
-      _disposeScrollNotificationObserver();
-      return;
-    }
-    if (_isInternalScrollableNotification(notification.context)) {
-      return;
-    }
-    if (!_scrollableNotificationIsFromSameSubtree(notification.context)) {
-      return;
+    switch (notification) {
+      case ScrollStartNotification() when _dataWhenToolbarShowScheduled != null:
+      case ScrollEndNotification() when _dataWhenToolbarShowScheduled == null:
+        break;
+      case ScrollEndNotification() when _dataWhenToolbarShowScheduled!.value != _value:
+        _dataWhenToolbarShowScheduled = null;
+        _disposeScrollNotificationObserver();
+      case ScrollNotification(:final BuildContext? context)
```

Comment:
Here, for instance, I missed the colon on first glance and thought it was just `case ScrollNotification(final BuildContext? context)` (doesn't help that `:` and `final` have the same color highlight). It took a minute to read carefully again before I understood what was going on.

Reply:
But maybe it's just me. I am new to Dart myself :)

Reply:
In this case (no pun intended), I'd prefer keeping it as-is.

<br>

The Dart documentation shows this concept in the [patterns overview](https://dart.dev/language/patterns#for-and-for-in-loops):

> …This allows you to simplify the variable pattern from something redundant like `key: key` to just `:key`

(and it goes into a bit more detail in the [pattern types](https://dart.dev/language/pattern-types#object) page)

Reply:
SGTM.